python/cudf_polars/cudf_polars/dsl/translate.py

```diff
@@ -64,16 +79,25 @@ def __exit__(self, *args: Any) -> None:
 
 @singledispatch
 def _translate_ir(
-    node: Any, visitor: NodeTraverser, schema: dict[str, plc.DataType]
+    node: Any,
+    visitor: NodeTraverser,
+    schema: dict[str, plc.DataType],
+    debug_mode: int = 0,
 ) -> ir.IR:
-    raise NotImplementedError(
-        f"Translation for {type(node).__name__}"
-    )  # pragma: no cover
+    e = f"Translation for {type(node).__name__}"
+    if debug_mode:
+        ERRORS.append(e)
+        return ir.ErrorNode(schema, e)
```

Comment:
Sorry I didn't catch this suggestion earlier. I'm not a huge fan of this global error object. For example, who is responsible for resetting it?

Since we're already modifying the signature of `_translate_ir`, how about we instead factor out both the `debug_mode` setting and the collection of errors into an object that we wrap around the `NodeTraverser`. 

Sketch:

```
class Translator:
    def __init__(self, visitor: NodeTraverser, *, debug_mode: bool):
        self.visitor = visitor
        self.debug_mode = debug_mode
        self.errors = []
```

Then we can add appropriate methods to `Translator`, possibly encapsulating the translation and collection of errors from the `NodeTraverser` (in case we raise on the rust side).

Afterwards in `callback.py` we can then just look at the `errors` attribute to see if we got any errors.

Does that make sense?